pkg/canary/deployment_controller.go

```diff
@@ -197,6 +197,11 @@ func (c *DeploymentController) ScaleFromZero(cd *flaggerv1.Canary) error {
 		if primary.Spec.Replicas != nil && *primary.Spec.Replicas > 0 {
 			replicas = primary.Spec.Replicas
 		}
+	} else if cd.Spec.AutoscalerRef != nil {
+		hpa, err := c.kubeClient.AutoscalingV2beta2().HorizontalPodAutoscalers(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
```

Comment:
we should try to fetch the v2 HPA first and if that fails, fall back to v2beta2 HPA. something similar to: https://github.com/fluxcd/flagger/blob/v1.22.1/pkg/canary/hpa_reconciler.go#L38-L50